baseplate/observers/tracing.py

```diff
@@ -539,18 +541,26 @@ class TraceQueueFullError(Exception):
 
 
 class SidecarRecorder(Recorder):
-    """Interface for recording spans to a POSIX message queue.
+    """Interface for recording spans to a message queue.
 
     The SidecarRecorder serializes spans to a string representation before
     adding them to the queue.
     """
 
-    def __init__(self, queue_name: str):
-        self.queue = MessageQueue(
-            "/traces-" + queue_name,
-            max_messages=MAX_QUEUE_SIZE,
-            max_message_size=MAX_SPAN_SIZE,
-        )
+    def __init__(self, queue_name: str, queue_type: QueueType = QueueType.POSIX):
+        if queue_type == QueueType.IN_MEMORY:
+            # Start a remote queue, which connects to a Thrift server
+            # that manages an in-memory queue
+            self.queue = RemoteMessageQueue(
+                "/traces-" + queue_name,
+                max_messages=MAX_QUEUE_SIZE,
+            )
+        else:
```

Comment:
As in other places you are encoding a default in 2 places, the init args and this if statement. If you change the default or add another queue type you will have to update loads of if statements. Better to just have a QueueType to implementation helper function and use that everywhere instead.